### PR TITLE
py-libEnsemble: add v1.0.0

### PR DESCRIPTION
**Pull request by shuds13**

### CODE REVIEW

**Review by manuelakuhn — COMMENTED** [review]
- `py-numpy`, `py-psutil`, `py-tomli` and `py-pyyaml` have a minimum version requirement in `py-libensemble@1.0.0`:
"numpy>=1.21", "psutil>=5.9.4", "tomli>=1.2.1", "pyyaml>=6.0"
(see [setup.py#l72](https://github.com/Libensemble/libensemble/blob/v1.0.0/setup.py#L72))

- "pydantic<2":
    ```
    - depends_on("py-pydantic@:2", type=("build", "run"), when="@0.10:")
    + depends_on("py-pydantic@:1", type=("build", "run"), when="@0.10:")
    ```

- Regarding the variants: since `py-libensemble@0.10.2` none of them is listed in `setup.py` anymore. `mpi` is at least mentioned in the documentation.
If they are not used anymore you can add a `when="..."` argument to the variant as well. If they still are, can you add a comment where this is documented or where they come from?

optional: 
- switch the order of `type` and `when` to make it consistent with the other python packages (`when` before `type`)

**Comment by shuds13** [comment]
I've added comments on the variants. We will make setup.py match up in future releases.

**Comment by shuds13** [comment]
@manuelakuhn

I'm hoping this is acceptable now if style checks pass.

**Review by manuelakuhn — APPROVED** [review]
LGTM. Thanks!

File: var/spack/repos/builtin/packages/py-libensemble/package.py [hunk 1]
```diff
@@ -40,32 +41,34 @@ class PyLibensemble(PythonPackage):
...
     variant("petsc4py", default=False, description="Install with petsc4py")
     variant("nlopt", default=False, description="Install with nlopt")
     variant("mpmath", default=False, description="Install with mpmath")
     variant("deap", default=False, description="Install with DEAP")
     variant("tasmanian", default=False, description="Install with tasmanian")
 
-    depends_on("py-numpy", type=("build", "run"))
-    depends_on("py-psutil", type=("build", "run"), when="@0.7.1:")
+    depends_on("py-numpy@1.21:", type=("build", "run"))
+    depends_on("py-psutil@5.9.4:", when="@0.7.1:", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-libensemble/package.py]:
```suggestion
    depends_on("py-numpy@1.21:", when="@1:", type=("build", "run"))
    depends_on("py-numpy", type=("build", "run"))
    depends_on("py-psutil@5.9.4:", when="@1:", type=("build", "run"))
    depends_on("py-psutil", when="@0.7.1:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-libensemble/package.py [hunk 2]
```diff
@@ -40,32 +41,34 @@ class PyLibensemble(PythonPackage):
...
     depends_on("py-setuptools", when="@0.10.2:", type="build")
     depends_on("py-setuptools", when="@:0.10.1", type=("build", "run"))
-    depends_on("py-pydantic@:2", type=("build", "run"), when="@0.10:")
-    depends_on("py-tomli", type=("build", "run"), when="@0.10:")
-    depends_on("py-pyyaml", type=("build", "run"), when="@0.10:")
+    depends_on("py-pydantic@:1", when="@0.10:", type=("build", "run"))
+    depends_on("py-tomli@1.2.1:", when="@0.10:", type=("build", "run"))
+    depends_on("py-pyyaml@6.0:", when="@0.10:", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-libensemble/package.py]:
```suggestion
    depends_on("py-tomli@1.2.1:", when="@1:", type=("build", "run"))
    depends_on("py-tomli", when="@0.10:", type=("build", "run"))
    depends_on("py-pyyaml@6.0:", when="@1:", type=("build", "run"))
    depends_on("py-pyyaml", when="@0.10:", type=("build", "run"))
```

**Review by manuelakuhn — CHANGES_REQUESTED** [review]
Sorry, I did not look properly and missed something:
you have added the version requirement on the dependencies now to all older releases as well. According to the `setup.py`s they should only be restricted for `py-libensemble@1.0.0` (and probably newer)

**Comment by shuds13** [comment]
@manuelakuhn I committed the suggestions. Is there still an issue?

**Review by manuelakuhn — APPROVED** [review]

**Comment by shuds13** [comment]
@manuelakuhn maybe someone can re-trigger the one failure. I'm guessing its some system thing.

**Comment by manuelakuhn** [comment]
@spackbot run pipeline

**Review by jlnav — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**